src/SFA.DAS.EmployerIncentives.Web.Tests/Controllers/CancelController/WhenSelectedApprentices.cs

```diff
@@ -0,0 +1,223 @@
+
+            // assert  
+            result.AccountId.Should().Be(_hashedAccountId);
+        }
+
+        [Test]
+        public async Task Then_legalEntityId_is_set_if_no_apprenticeships_selected()
+        {
+            // arrange
+            var selected = new SelectApprenticeshipsRequest()
+            {
+                AccountId = _hashedAccountId,
+                AccountLegalEntityId = _hashedAccountLegalEntityId,
+                SelectedApprenticeships = new List<string>()
+            };
+
+            // act
+            var result = ((ViewResult)await _sut.SelectApprenticeships(selected)).Model as SelectApprenticeshipsViewModel;
+
+            // assert 
+            result.AccountLegalEntityId.Should().Be(_hashedAccountLegalEntityId);
+        }
+
+        [Test]
+        public async Task Then_organisationName_is_set_if_no_apprenticeships_selected()
+        {
+            // arrange
+            var selected = new SelectApprenticeshipsRequest()
+            {
+                AccountId = _hashedAccountId,
+                AccountLegalEntityId = _hashedAccountLegalEntityId,
+                SelectedApprenticeships = new List<string>()
+            };
+
+            // act
+            var result = ((ViewResult)await _sut.SelectApprenticeships(selected)).Model as SelectApprenticeshipsViewModel;
+
+            // assert 
+            result.OrganisationName.Should().Be(_organisationName);
+        }
+
+
+        [Test]
+        public async Task Then_should_display_a_list_of_apprentices()
+        {
+            // arrange
+            var selected = new SelectApprenticeshipsRequest()
+            {
+                AccountId = _hashedAccountId,
+                AccountLegalEntityId = _hashedAccountLegalEntityId,
+                SelectedApprenticeships = new List<string>()
+            };
+
+            // act
+            var result = ((ViewResult)await _sut.SelectApprenticeships(selected)).Model as SelectApprenticeshipsViewModel;
+
+            // assert 
+            result.ApprenticeshipIncentives.Should()
+                .BeEquivalentTo(_apprenticeshipIncentiveData,
+                    opt => opt
+                        .Excluding(x => x.Id)
+                        .Excluding(x => x.DisplayName)
+                );
+        }
+
+        [Test]
+        public async Task Then_should_have_apprentices_ordered_by_last_name()
+        {
+            // arrange
+            var selected = new SelectApprenticeshipsRequest()
+            {
+                AccountId = _hashedAccountId,
+                AccountLegalEntityId = _hashedAccountLegalEntityId,
+                SelectedApprenticeships = new List<string>()
+            };
+
+            // act
+            var result = ((ViewResult)await _sut.SelectApprenticeships(selected)).Model as SelectApprenticeshipsViewModel;
+
+            // assert 
+            result.ApprenticeshipIncentives.Should().BeInAscendingOrder(x => x.LastName);
```

Comment:
AC1 of ticket EI-938 says that the apprentices should be sorted by first name, last name, then by ULN - for consistency